lib/url.c

```diff
@@ -2355,21 +2355,10 @@ static CURLcode parse_proxy(struct Curl_easy *data,
 static CURLcode parse_proxy_auth(struct Curl_easy *data,
                                  struct connectdata *conn)
 {
-  char proxyuser[MAX_CURL_USER_LENGTH]="";
-  char proxypasswd[MAX_CURL_PASSWORD_LENGTH]="";
+  char *proxyuser = data->set.str[STRING_PROXYUSERNAME];
+  char *proxypasswd = data->set.str[STRING_PROXYPASSWORD];
```

Comment:
Are there any guarantees that both are non-NULL? If not, that could make `Curl_urldecode` crash.

Reply:
I figured the tests would crash and burn then... =)

Reply:
The function `parse_proxy_auth()`  is only called if `conn->bits.proxy_user_passwd` is set, which presumably never happens if they are NULL.

Reply:
Hm, the password can actually still be NULL...

Reply:
Adding checks